### PR TITLE
addressing etree error under py3.9, resolves #535

### LINKED ISSUE #535: AttributeError: 'xml.etree.ElementTree.Element' object has no attribute 'getchildren'
since Py 3.9 the function getchildren() in etree seems to be removed (was deprecated since long before, AFAIK)
Affects AMI corpus preparation
```
(py3) [jtrmal@test4 ami_lhotse]$ lhotse prepare ami --partition full-corpus-asr --normalize-text  /expscratch/mwiesner/miniscale_2022_data/Data/amicorpus .
2022-01-13 12:01:58,314 INFO [ami.py:554] Parsing AMI annotations
Traceback (most recent call last):
  File "/home/hltcoe/jtrmal/miniconda3/envs/py3/bin/lhotse", line 33, in <module>
    sys.exit(load_entry_point('lhotse', 'console_scripts', 'lhotse')())
  File "/home/hltcoe/jtrmal/miniconda3/envs/py3/lib/python3.9/site-packages/click/core.py", line 1128, in __call__
    return self.main(*args, **kwargs)
  File "/home/hltcoe/jtrmal/miniconda3/envs/py3/lib/python3.9/site-packages/click/core.py", line 1053, in main
    rv = self.invoke(ctx)
  File "/home/hltcoe/jtrmal/miniconda3/envs/py3/lib/python3.9/site-packages/click/core.py", line 1659, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/hltcoe/jtrmal/miniconda3/envs/py3/lib/python3.9/site-packages/click/core.py", line 1659, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/hltcoe/jtrmal/miniconda3/envs/py3/lib/python3.9/site-packages/click/core.py", line 1395, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/hltcoe/jtrmal/miniconda3/envs/py3/lib/python3.9/site-packages/click/core.py", line 754, in invoke
    return __callback(*args, **kwargs)
  File "/home/hltcoe/jtrmal/lhotse/lhotse/bin/modes/recipes/ami.py", line 47, in ami
    prepare_ami(
  File "/home/hltcoe/jtrmal/lhotse/lhotse/recipes/ami.py", line 565, in prepare_ami
    annotations = parse_ami_annotations(annotations_dir, normalize_text)
  File "/home/hltcoe/jtrmal/lhotse/lhotse/recipes/ami.py", line 342, in parse_ami_annotations
    assert len(seg.getchildren()) == 1, "Multiple child segments found"
AttributeError: 'xml.etree.ElementTree.Element' object has no attribute 'getchildren'
```

### PR DESCRIPTION
**Pull request by jtrmal**

### CODE REVIEW

File: lhotse/recipes/ami.py [hunk 1]
```diff
@@ -339,8 +339,8 @@ def _parse_href(href, wid_to_word):
                     continue
                 start_time = float(seg.attrib["transcriber_start"])
                 end_time = float(seg.attrib["transcriber_end"])
-                assert len(seg.getchildren()) == 1, "Multiple child segments found"
-                seg_child = seg.getchildren()[0]
+                assert len(list(seg)) == 1, "Multiple child segments found"
+                seg_child = list(seg)[0]
```
Comment (csukuangfj) [inline lhotse/recipes/ami.py:343]: How about
```python3
seg_child = next(iter(seg))
```
?

File: lhotse/recipes/ami.py [hunk 2]
```diff
@@ -339,8 +339,8 @@ def _parse_href(href, wid_to_word):
                     continue
                 start_time = float(seg.attrib["transcriber_start"])
                 end_time = float(seg.attrib["transcriber_end"])
-                assert len(seg.getchildren()) == 1, "Multiple child segments found"
-                seg_child = seg.getchildren()[0]
+                assert len(list(seg)) == 1, "Multiple child segments found"
```
Comment (csukuangfj) [inline lhotse/recipes/ami.py:342]:
```suggestion
                assert len(seg) == 1, "Multiple child segments found"
```

Please see the doc at https://docs.python.org/3/library/xml.etree.elementtree.html

> Element objects also support the following sequence type methods for working with subelements: `__delitem__()`, `__getitem__()`, `__setitem__()`, `__len__().`

**Comment by jtrmal** [comment]
yeah, those changes will definitely make it nicer, will fix
y.

**Comment by pzelasko** [comment]
Thanks!

**Pull request merged**